core/Pos/Arbitrary/Core.hs

```diff
@@ -443,13 +442,14 @@ instance Arbitrary G.GenesisCoreData where
                          <$> arbitrary
                          <*> pure (wordILen - a)
                          <*> arbitrary
-                , pure $ G.ExponentialStakes wordILen
+                , pure $ G.ExponentialStakes wordILen (Types.mkCoin 1024)
```

Comment:
Please read my first comment in this discussion. It answers both of your questions. In particular:

>We also require that no genesis balance is lower than sum of weights. 

If you set to `3` or to `1000` genesisUtxo won't be created because there's a balance that is less than `bootDustThd`.